server/static/js/admin/controllers.js

```diff
@@ -364,14 +378,28 @@ app.controller("CourseListCtrl", ['$scope', 'Course',
     $scope.courses = Course.query({});
   }]);
 
-  app.controller("CourseAssignmentsCtrl", ['$scope', '$http', 'Assignment', 'Course', '$stateParams',
-    function($scope, $http, Assignment, Course, $stateParams) {
+  app.controller("CourseAssignmentsCtrl", ['$scope', '$http', 'Assignment', 'Course', '$stateParams', '$window',
+    function($scope, $http, Assignment, Course, $stateParams, $window) {
     $scope.course = Course.get({id: $stateParams.courseId});
-     Course.assignments({
-      id: $stateParams.courseId
-     },function(response) {
-       $scope.assignments = response
-     });
+    $scope.reloadView = function() {
+       Course.assignments({
+        id: $stateParams.courseId
+       },function(response) {
+         $scope.assignments = response
+       });
+     }
+
+     $scope.delete = function(assign) {
+       Assignment.delete({
```

Comment:
Ideally we'd avoid repeated code - but it's ok. 

Same thing about the deletion (warning)